### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by SebastienPeillet**
- [x] help icon : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1820514186
- [x] warning when compiling doc : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1820514186
- [x] life cycle of the boat icon : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1820566245
- [x] init export progress bar : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1820581448
- [x] invalid error messages : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1820586245
- [x] validate action is not properly working : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1831523323
- [x] no warning messages for newly created project : https://github.com/hytechimaging/sammo-boat/pull/197#issuecomment-1831536346
- [x] error when creating project on Ubuntu 22 (QGIS 3.22, Python 3.10) > outdated, the next version will be for QGIS >= 3.28

### CODE REVIEW

**Comment by pblottiere** [comment]
The `Help` icon is not displayed in sammo toolbar:

image

And I have a warning when compiling the documentation:

``` console
reading sources... [100%] session
/home/pblottiere/devel/hytech/github/sammo-boat/doc/source/session.rst:4: WARNING: duplicate label session, other instance in /home/pblottiere/devel/hytech/github/sammo-boat/doc/source/session.rst
looking for now-outdated files... none found
```

**Comment by pblottiere** [comment]
As soon as a new line is added in environment table with `routeType = prospection`, the boat icon is always green whatever is happening next:

image

File: doc/source/interface.rst [hunk 1]
```diff
@@ -57,11 +57,18 @@ This button allows to open the settings dialog, to configure administrators tabl
...
+fulfill the session metadata. In the `Survey` table, `survey` attribute list
+values come from the `SurveyType` table, `shipName` attribute list values come
+from the `Boat` table.
 
-`Boat` and `Plateform` buttons open their tables dialog as each can contains more
-than one entity.
+`SurveyType`, `Transect`, `Boat` and `Plateform` buttons open their tables
+dialog as each can contains more than one entity.
+
+Espiecially for `Transect` table, it is possible to import a linear layer
```
Comment (pblottiere) [inline doc/source/interface.rst:68]:
```suggestion
Especially for `Transect` table, it is possible to import a linear layer
```

File: doc/source/interface.rst [hunk 2]
```diff
@@ -201,18 +210,16 @@ A status will be assigned automatically to each entity.
 The first status will be ``Begin``, then ``Add`` status will be created for the
 next entities until user changes the ``routeType`` attribute. User creates
 ``Add`` entities if the environmental variables change during the route. When the
-user changes the ``routeType`` attributes, it duplicates the previous entity
-(``Begin`` or ``Add`` status) and assignes the ``End`` status to the duplica.
-Environment status are check before the validation. A dialog should pop-up if 
-there is a missing ``Start`` / ``End`` record to inform user. Use the duplicate
-action to fix it (cf. :ref:`duplicateaction`).
+user changes the ``routeType`` attributes, the status attribute will be update to
```
Comment (pblottiere) [inline doc/source/interface.rst:213]:
```suggestion
user changes the ``routeType`` attributes, the status attribute will be updated to
```

File: src/core/database.py [hunk 3]
```diff
@@ -83,14 +84,26 @@ def init(self, directory: str) -> bool:
             self._createFieldsForGpsTable(), GPS_TABLE, QgsWkbTypes.Point
         )
 
-        self._createTable(self._fieldsObserver(), OBSERVERS_TABLE)
-
         # administrator table
+        self._createTable(self._fieldsObserver(), OBSERVERS_TABLE)
+        self._populateTable(SPECIES_TABLE, "observers.csv")
```
Comment (pblottiere) [inline src/core/database.py:89]:
```suggestion
        self._populateTable(OBSERVERS_TABLE, "observers.csv")
```

File: src/core/layers/survey_type.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2022 Hytech Imaging"
```
Comment (pblottiere) [inline src/core/layers/survey_type.py:4]:
```suggestion
__copyright__ = "Copyright (c) 2023 Hytech Imaging"
```

File: src/core/session.py [hunk 5]
```diff
@@ -463,170 +489,33 @@ def validate(self, merge=False):
...
-            ):
-                errors.append(
-                    f"Missing END code for effortGroup {effortId} (after "
-                    f"{max(statusCodes).toPyDateTime().isoformat()} record)"
-                )
-        if errors:
-            errors.append("Please, resolve errors before validation")
-            QMessageBox.warning(
-                None, "Errors detected in effort status", "\n".join(errors)
+        def validateFeatures(selectedLayer: QgsVectorLayer):
```
Comment (pblottiere) [inline src/core/session.py:494]:
```suggestion
        def validateFeatures(selectedLayer: QgsVectorLayer) -> None:
```

File: src/gui/merge.py [hunk 6]
```diff
@@ -90,12 +102,212 @@ def sessionMerged(self) -> None:
...
+
+    def run(self) -> bool:
+        try:
+            self.merge()
+        except Exception as e:
+            self.errorMsg = ",".join([str(i) for i in e.args])
+            return False
+        return True
+
+    def merge(self):
```
Comment (pblottiere) [inline src/gui/merge.py:156]:
```suggestion
    def merge(self) -> None:
```

**Comment by SebastienPeillet** [comment]
> As soon as a new line is added in environment table with `routeType = prospection`, the boat icon is always green whatever is happening next:
> 
> image

Weird, it works on my side. Is it better now ?

**Comment by SebastienPeillet** [comment]
> The `Help` icon is not displayed in sammo toolbar:
> 
> image
> 
> And I have a warning when compiling the documentation:
> 
> ```
> reading sources... [100%] session
> /home/pblottiere/devel/hytech/github/sammo-boat/doc/source/session.rst:4: WARNING: duplicate label session, other instance in /home/pblottiere/devel/hytech/github/sammo-boat/doc/source/session.rst
> looking for now-outdated files... none found
> ```

Weird, it works on my side... Is it better now ?

**Comment by SebastienPeillet** [comment]
> As soon as a new line is added in environment table with `routeType = prospection`, the boat icon is always green whatever is happening next:
> 
> image

It seems it didn't work as expected [since 1,5 years](https://github.com/hytechimaging/sammo-boat/commit/b3040e84e51ae6e44c4b76588f6f5bde5ae79292#diff-7fb8390f00b9aa9417fc3e222addc5a9e4f61c39d60b474f01190f6acf138e0e) ^^ I fix this asap

**Comment by pblottiere** [comment]
>> The Help icon is not displayed in sammo toolbar:

> Weird, it works on my side... Is it better now ?

Yes it's back now :tada:

**Comment by pblottiere** [comment]
On Windows with QGIS 3.34, the attribute table is not docked when creating a project:

sammob_window_detached

**Comment by pblottiere** [comment]
With QGIS 3.34 (tested on ArchLinux and Windows), the *Validate* action is not working properly. Actually, background of cells in attribute tables is not updated. And only one cell is updated after an *Export* action:

image

**Comment by pblottiere** [comment]
> On Windows with QGIS 3.34, the attribute table is not docked when creating a project:

(once manually docked, it doesn't happen for new projects)

**Comment by pblottiere** [comment]
When I create a project, I can add environment/sightings without warnings even if I didn't fill administrator tables (*survey*, ...). But when I reopen a sammo project after closing QGIS, I have warnings:

warning

**Comment by pblottiere** [comment]
I tried to use the plugin on Ubuntu 22 (QGIS 3.22, Python 3.10) and I have the next error:

image

**Comment by SebastienPeillet** [comment]
> I tried to use the plugin on Ubuntu 22 (QGIS 3.22, Python 3.10) and I have the next error:
> 
> image

This method appeared in 3.26 (cf. https://github.com/qgis/QGIS/commit/a578ecb33ad24517b19fc82e2dc4752526926847)

**Comment by SebastienPeillet** [comment]
> With QGIS 3.34 (tested on ArchLinux and Windows), the _Validate_ action is not working properly. Actually, background of cells in attribute tables is not updated. And only one cell is updated after an _Export_ action:
> 
> image

It seems that there is weird behaviour on 3.34 when a field used in conditionalStyle is hidden...

**Pull request merged**